pkg/objectmanager/objectmanager.go

```diff
@@ -0,0 +1,151 @@
+package objectmanager
+
+import (
+	"nimona.io/pkg/context"
+	"nimona.io/pkg/crypto"
+	"nimona.io/pkg/errors"
+	"nimona.io/pkg/exchange"
+	"nimona.io/pkg/log"
+	"nimona.io/pkg/object"
+	"nimona.io/pkg/peer"
+	"nimona.io/pkg/sqlobjectstore"
+)
+
+var (
+	requestType = new(Request).GetType()
+)
+
+type (
+	Requester interface {
+		Request(
+			ctx context.Context,
+			hash object.Hash,
+			peer *peer.Peer,
+		) (*object.Object, error)
+	}
+	manager struct {
+		store    *sqlobjectstore.Store
+		exchange exchange.Exchange
+	}
+	Option func(*manager)
+)
+
+func New(
+	ctx context.Context,
+	opts ...Option,
+) Requester {
+	m := &manager{}
+
+	for _, opt := range opts {
+		opt(m)
+	}
+
+	logger := log.
+		FromContext(ctx).
+		Named("objectmanager").
+		With(
+			log.String("method", "objectmanager.New"),
+		)
+
+	subs := m.exchange.Subscribe(
+		exchange.FilterByObjectType(requestType),
+	)
+
+	go func() {
+		if err := m.handleObjects(subs); err != nil {
+			logger.Error("handling object requests failed", log.Error(err))
+		}
+	}()
+
+	return m
+}
+
+func (m *manager) Request(
+	ctx context.Context,
+	hash object.Hash,
+	pr *peer.Peer,
+) (*object.Object, error) {
+	var obj *object.Object
+	errCh := make(chan error)
+
+	sub := m.exchange.Subscribe(
```

Comment:
You probably want to `defer sub.Cancel()` so the queue gets destroyed when we return.